app/controllers/accounts_controller.rb

```diff
@@ -32,13 +33,39 @@ def show
 
       format.json do
         expires_in 3.minutes, public: !(authorized_fetch_mode? && signed_request_account.present?)
+
+        record_reach!
+
         render_with_cache json: @account, content_type: 'application/activity+json', serializer: ActivityPub::ActorSerializer, adapter: ActivityPub::Adapter
       end
     end
   end
 
   private
 
+  def record_reach!
+    unless authorized_fetch_mode?
+      AccountReachFilter.where(account_id: @account.id).delete_all
+      return
+    end
```

Comment:
Is this so if you turn off authorized fetch mode all the bloom filters are deleted? This seems like it should be a CLI task?

Reply:
That's it. I'm not sure this should be a CLI task, not deleting the bloom filter opens you to miss recipients in your delivery of `Delete` activities.

Reply:
At the very least then, this should be checked and done on application startup, not at request time.